jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/DataIteratorsTests.java

```diff
@@ -99,4 +107,37 @@ public void singleSelectDataIteratorTest() {
         singleSelectTableIterator.is().unchecked(0);
         singleSelectTableIterator.is().checked(1);
     }
+
+    @Test(description = "Test checks data iterator parameter : No data text")
+    public void noDataTextDataIteratiorTest() {
+        noDataTextDataIterator.show();
+        noDataTextDataIterator.has().text(containsString("Sorry, there's no any items data..."));
+        //The default text is "No data available".
+        //Here we also check the parameter no-data-text="..." that allows to set custom text value for the parameter.
+    }
+
+    @Test(description = "Test checks data iterator parameter : No results text")
+    public void noResultsTextDataIteratiorTest() {
+        noResultsTextDataIteratorSearchField.show();
+        noResultsTextDataIteratorSearchField.clearTextField();
+        noResultsTextDataIteratorDataItemList.is().notEmpty();
+        noResultsTextDataIteratorSearchField.clearAndTypeText("abcd");
+        noResultsTextDataIterator.has().text(containsString("Sorry, nothing found :("));
+        //The default text is "No matching records found".
+        //Here we also check the parameter no-results-text="..." that allows to set custom text value for the parameter.
+    }
+
+    @Test(description = "Test checks data iterator parameter : Loading text")
+    public void loadingTextDataIteratiorTest() {
+        loadingTextDataIterator.show();
+        loadingTextDataIterator.has().text(containsString("Items are loading, please, wait a little bit..."));
+        //The default text is "Loading items...".
+        // Here we also check the parameter loading-text="..." that allows to set custom text value for the parameter.
+    }
+
+    @Test(description = "Test checks data iterator parameter : Group by")
+    public void groupByDataIteratorTest() {
+        groupedByDataIterator.show();
+        groupedByDataIterator.has().size(greaterThan(1));
+    }
```

Comment:
Please, try to find a way how to check the main functionality here: 'groupig' items by chosen parameter
`desserts: [
        {
          name: 'Frozen Yogurt',
          calories: 159,
          fat: 6.0,
          carbs: 24,
          protein: 4.0,
          sodium: 87,
          calcium: '14%',
          iron: '1%',
        },
        {
          name: 'Ice cream sandwich',
          calories: 237,
          fat: 9.0,
          carbs: 37,
          protein: 4.3,
          sodium: 129,
          calcium: '8%',
          iron: '1%',
        },
        {
          name: 'Eclair',
          calories: 262,
          fat: 16.0,
          carbs: 23,
          protein: 6.0,
          sodium: 337,
          calcium: '6%',
          iron: '7%',
        },
        {
          name: 'Cupcake',
          calories: 305,
          fat: 3.7,
          carbs: 67,
          protein: 4.3,
          sodium: 413,
          calcium: '3%',
          iron: '8%',
        },
      ],`

Reply:
Implemented DataProvider solution in a little bit different way, please check